repo_tests/fixture_fixed_path_teardown_guard.py

```
@@ -340,17 +257,70 @@ def _derived_names(func: ast.FunctionDef | ast.AsyncFunctionDef) -> Set[str]:
+    params = _scope_params(scope)
+    by_name = _assignments_by_name(scope)
+    seeds = (inherited - params) | (params & _UNIQUE_SOURCE_PARAMS)
     return _withdraw_partly_fixed(by_name, _reachably_derived(by_name, seeds))
 
 
+def _derived_names(func: ast.FunctionDef | ast.AsyncFunctionDef) -> Set[str]:
+    """``_scope_derived_names`` for a fixture, which inherits nothing."""
+    return _scope_derived_names(func, set())
+
+
+def _record_reached(reached: Dict[int, ReachedScope], target: Scope, guarded: bool, inherited: Set[str]) -> bool:
```

Ruff (0.16.3)

[warning] Boolean-typed positional argument in function definition

(FBT001)